jenkins/Pipeline.gy

```diff
@@ -2,7 +2,7 @@ pipeline {
     agent { label 'pbench' }
     environment {
         // Environment variables for various `jenkins/run` invocations.
-        EXTRA_PODMAN_SWITCHES="--pull=always -e=COV_REPORT_XML=yes"
+        EXTRA_PODMAN_SWITCHES="--pull=always -e=COV_REPORT_XML=yes --network host"
```

Comment:
Not necessarily objecting to making this the default (I think we probably either need it everywhere, or at least it wouldn't hurt); but why add here *and* in `build.sh`? Yes, we might run `build.sh` "offline", but then do we really need to change the default here?

Reply:
The reason for doing it in `build.sh` is that the invocation of the `alembic-check` needs to have `--network host` (while most invocations of `jenkins/run` actually don't, although AFAIK it doesn't hurt to have it for them), so, in order to make `build.sh` useful _outside_ the CI environment, we need to specify it in the invocation in there.  Moreover, the invocation of `alembic-check` does _not_ need the `COV_REPORT_XML` environment variable (and it's probably better off without it), so overriding the default value of `EXTRA_PODMAN_SWITCHES` in `build.sh` is arguably a plus.

That said, I guess we don't need to add `--network host` to the pipeline default, since I'm not aware of any other uses of `EXTRA_PODMAN_SWITCHES` which need the switch.  However, as I said, I don't think adding it hurts anything, and having it there will cover any future cases of cooperating container invocations.

One note is that the redefinition of `EXTRA_PODMAN_SWITCHES` in `build.sh` overrides any pull policy set in the outer environment (like the pipeline default).  In the CI, this shouldn't have any effect, because `build.sh` is invoked via `jenkins/run` in a pipeline stage, and so _that_ invocation will have pulled a fresh copy of the container, per the pipeline default `EXTRA_PODMAN_SWITCHES` value, so _not_ having `--pull always` defined for the inner invocation is actually a plus there, as well, since it avoids a pointless pull when running under the CI.  (For use of `build.sh` outside the CI, whether/how `EXTRA_PODMAN_SWITCHES` is defined in the developer's environment (and whether a fresh container image needs to be pulled) is up to the developer.)

Reply:
> as I said, I don't think adding it hurts anything

And, apparently, I was wrong!  Adding the `--network host` causes `/sys/kernel` inside the container to be overmapped with an empty `tmpfs` file system, which breaks the Agent legacy `datalog` `sysfs` test by preventing it from finding the `/sys/kernel/vmcoreinfo` file!

So, I've moved the switch out of the global pipeline default and added it only to the invocation of the linters' run.  😝